### ISSUE TITLE
str formatting of views

### ISSUE THREAD

**Issue by hagberg** [networkx/networkx#2611]
I prefer seing

```python
>>> print(G.edges())
[(1, 2), (2, 3)])
```
over
```python
>>> print(G.edges())
EdgeView([(1, 2), (2, 3)])
```
as the current 2.0 views show.

One goal of of ``__str__`` is to make a user-readable representation of the data and the user probably doesn't care much that this is an EdgeView.   If they do then they can use ``__repr__`` right?

Opinions?  Does that change break anything?  It seems more consistent with networkx-1. output

**Comment by dschult** [comment]
Unfortunately, ipython and probably other systems use ```repr``` to show values of the expression. So without the ```print``` the more detailed output appears. 

    >>> print(G.edges())
    [(1, 2), (2, 3)]
    >>> G.edges()
    EdgeView([(1, 2), (2, 3)])

The repr output does not fulfill the Python doc request that ```repr``` provide sufficient information to recreate the object "in an appropriate environment".  That would require  even more output.

Should we make ```repr``` produce list-like output and forego the view name?
That allows ```repr``` and ```str``` to be the same. We gain from simple output. We lose the hint that this is more than a list.

**Comment by hagberg** [comment]
It makes sense to have an obvious way to get what the underlying object is - repr.  At least for programmers.  I think this is OK.

**Comment by dschult** [comment]
addressed by #2613
